library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java

```diff
@@ -333,7 +333,6 @@ private void initAttachmentUI() {
     }
 
     // endregion
-    // TODO: the name of this method is weird (progres..)? perhaps captureMedia?
     public void captureMedia(int requestCode, int resultCode, Intent data) {
```

Comment:
The method is really complicated. And new `clearInsertedMediaStore` calls make it even harder to read.
I see at least few things which might be simplified:

1. This doesn't throw any checked exception:
```java
String path = imageUri.getPath();
file = new File(path);
```
But `getPath` can return null, so we can wrap it into util function:
```java
@Nullable
static File getFileFromUri(Uri uri){
  String path = imageUri.getPath();
  if(path == null)
    return null;
  else
    return new File(path);
}
```
2. The same about second try/catch, there is no checked exceptions

So try/catches could be eliminated and wrapped into some another utility functions and eventually the method might look like this:
```java
public void captureMedia(int requestCode, int resultCode, Intent data) {
	if (requestCode == Constant.CAPTURE_IMAGE_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
	  Uri uri = UriUtils.getUri(data);
	  if(uri != null) messageInputController.progressCapturedMedia(getContext(), uri, false);
	}
	clearInsertedMediaStore();
}

Try/catches reduce readability and code flow significantly. If there're more that two try in a method it's a sign that something should be extracted and moved into helper functions.
```


```

Reply:
@elevenetc , currently sdk is keeping images and videos from camera after sending attachment message or not.
Then... `Skype` clears them by created in app after sending attachments.
`Instagram` is keeping them.
Which do you think is better?

Reply:
I would not use `Skype` and `Instagram` as reference messaging apps 😛 

Checked `Telegram` and it doesn't delete photos. Can't check `Whatsapp`, it's down now. Let's then also keep: it's easier to implement and doesn't require support. Also makes sense actually: users are owners of their photos, so they should have originals.

And by the way, `Telegram` creates it's own directory and inside keep all files sorted by file type. Also puts cache of all other files (from Glide in our case). That's a nice thing.

![Mon_Nov_11_16:30:10_CET_2019](https://user-images.githubusercontent.com/1069159/68599330-b53a5980-04a0-11ea-98ad-766c8d9fddc9.png)

Reply:
I guessed you would send Telegram case screenshot since you are `Telegram` fan. 😛
Got it. I think it would be better.